src/main/java/duke/io/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke.io;
+
+import duke.io.Command.CommandName;
+import duke.DukeException;
+
+import java.util.Scanner;
+
+/**
+ * Parser class intended to be used to parse user input to extract the correct command.
+ */
+public class Parser {
+
+    /**
+     * Retrieves the next line of input, disregarding extra whitespaces.
+     *
+     * @param sc The scanner object to retrieve the next line of input from
+     * @return The next line of input after trimming whitespace from both ends
+     */
+    public static String getInput(Scanner sc) {
+        return sc.nextLine().trim();
+    }
+
+    /**
+     * Parses a line of given user input to extract the relevant/correct command.
+     *
+     * @param userInput A String containing a line of input to parse as a single command
+     * @return A Command object containing an identifier of the command and its arguments
+     * @throws DukeException When given input cannot be parsed successfully
+     */
+    public static Command parse(String userInput) throws DukeException {
```

Comment:
I think this method is too long? Maybe can consider using helper methods to simplify the logic?